src/components/EmailForm.tsx

```diff
@@ -7,21 +7,54 @@ import { CheckboxSmall, Input } from './FormComponents';
 import { saveEmail } from '../lib/api';
 import { EmailUserInput } from '../types/UserInput';
 
+const getErrorMessage = (code: string, message: string) => {
+  switch (code) {
+    case 'duplicate_parameter':
+      return 'Tento email už v databáze existuje';
+    case 'invalid_parameter':
+      return 'Nesprávny formát emailovej adresy';
+    default:
+      return `Chyba: ${message}`;
+  }
+};
+
 export interface EmailFormProps {
-  name: string;
+  applicantFullName: string;
+  deadline: string;
+  formName: string;
 }
-export const EmailForm = ({ name }: EmailFormProps) => {
-  const [savingInProgress, setSavingInProgress] = useState<boolean>(false);
+export const EmailForm = ({
+  applicantFullName,
+  deadline,
+  formName,
+}: EmailFormProps) => {
   const [savedWithNewsletter, setSavedWithNewsletter] = useState<boolean>(
     undefined,
   );
-  const [email, setEmail] = useState<string>('');
+  const [savedEmail, setSavedEmail] = useState<string>('');
+
+  const handleSubmit = async ({ email, newsletter }, { setFieldError }) => {
+    const [firstName, ...lastName] = applicantFullName.split(' ');
+    const { id, code, message } = await saveEmail(email, {
+      firstname: firstName,
+      lastname: lastName.join(' '),
+      newsletter: !!newsletter,
+      deadline,
+      form: formName,
+    });
+    if (id) {
+      setSavedWithNewsletter(newsletter);
+      setSavedEmail(email);
+    } else {
+      setFieldError('email', getErrorMessage(code, message));
+    }
```

Comment:
Tu treba este zavolat `setSubmitting(false)` z formiku https://jaredpalmer.com/formik/docs/guides/form-submission

Reply:
To tam netreba :)